codalab/server/bundle_manager.py

```diff
@@ -692,6 +693,25 @@ def _get_staged_bundles_to_run(self):
                     bundle,
                     {'state': State.FAILED, 'metadata': {'failure_message': failure_message}},
                 )
+            elif bundle.metadata.request_queue:
+                # For those bundles that were requested to run on a private worker, if the private worker does not exist
+                # temporarily, we dispatch those bundles until the private worker connected to CodaLab server.
+                request_queue = bundle.metadata.request_queue
+                if request_queue:
+                    tagm = re.match('tag=(.+)', request_queue)
```

Comment:
We should have a utility function for this (there must be other places where we have this logic).

Reply:
This could also return `None`, don't crash on that.